tools/generators/normalize-package-dependencies/index.ts

```diff
@@ -0,0 +1,198 @@
+import {
+  Tree,
+  formatFiles,
+  getProjects,
+  ProjectConfiguration,
+  updateJson,
+  joinPathFragments,
+  readJson,
+  logger,
+  createProjectGraphAsync,
+  ProjectGraph,
+} from '@nrwl/devkit';
+
+import { NormalizePackageDependenciesGeneratorSchema } from './schema';
+import { PackageJson } from '../../types';
+import * as chalk from 'chalk';
+
+type ProjectIssues = { [projectName: string]: { [depName: string]: string } };
+
+type NormalizedSchema = ReturnType<typeof normalizeOptions>;
+
+export default async function (tree: Tree, schema: NormalizePackageDependenciesGeneratorSchema) {
+  const normalizedOptions = normalizeOptions(tree, schema);
+
+  const graph = await createProjectGraphAsync();
+
+  const filters = getActiveFilters(normalizedOptions);
+  const projects = getProjects(tree);
+  const issues: ProjectIssues = {};
+
+  projects.forEach(projectConfig => {
+    if (!shouldBeProjectProcessed(projectConfig, filters)) {
+      return;
+    }
+
+    if (normalizedOptions.verify) {
+      const foundIssues = getPackageJsonDependenciesIssues(tree, projectConfig, graph);
+
+      if (foundIssues) {
+        issues[projectConfig.name!] = foundIssues;
+      }
+      return;
+    }
+
+    normalizePackageJsonDependencies(tree, projectConfig, graph);
+  });
+
+  reportPackageJsonDependenciesIssues(issues);
+
+  await formatFiles(tree);
+}
+
+function getActiveFilters(options: NormalizedSchema) {
+  const active: Partial<
+    Record<keyof Omit<NormalizedSchema, 'verify'>, (projectConfig: ProjectConfiguration) => boolean>
+  > = {};
+
+  if (options.projectType !== 'any') {
+    active.projectType = (projectConfig: ProjectConfiguration) => {
+      return projectConfig.projectType === options.projectType;
+    };
+  }
+  if (options.tag) {
+    active.tag = (projectConfig: ProjectConfiguration) => {
+      return Array.isArray(projectConfig.tags) && projectConfig.tags.includes(options.tag!);
+    };
+  }
+
+  const activeCount = Object.keys(active).length;
+
+  return {
+    hasActive: activeCount > 0,
+    hasActiveMultiple: activeCount > 1,
+    filters: active,
+  };
+}
+
+function shouldBeProjectProcessed(projectConfig: ProjectConfiguration, filters: ReturnType<typeof getActiveFilters>) {
+  const filterPredicates = Object.values(filters.filters);
+
+  if (filters.hasActiveMultiple) {
+    return filterPredicates.every(predicate => predicate(projectConfig));
+  }
+
+  if (filters.hasActive) {
+    return filterPredicates.some(predicate => predicate(projectConfig));
+  }
```

Comment:
is the change in behaviour of `every` and `some` necessary? It seems that just using `every` is enough.

`hasActive` is only triggered when there is only one filter present and in that case `some` and `every` will work in the same way?

Reply:
good catch, yeah lets unify this -> also we can remove `hasActiveMultiple` completely. ty!

Reply:
https://github.com/microsoft/fluentui/pull/28382/commits/e23ee5e6eb2e8c7d1a11bfea4358a62fe526eb78